ludwig/schema/preprocessing.py

```diff
@@ -26,6 +26,13 @@ class PreprocessingConfig(schema_utils.BaseMarshmallowConfig):
         parameter_metadata=PREPROCESSING_METADATA["sample_size"],
     )
 
+    eval_steps: float = schema_utils.NonNegativeInteger(
```

Comment:
I don't think this preprocessing implementation is the right approach.

`eval_steps` should be a trainer parameter and used within `run_evaluation()/batch_evaluate()`.